Simple Incremental/Assets/Scripts/CharacterHealthEffect.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(CharacterHealth))]
+public class CharacterHealthEffect : MonoBehaviour
+{
+    CharacterHealth characterHealth;
+    ParticleSystem system;
+
+    void Awake()
+    {
+        characterHealth = GetComponent<CharacterHealth>();
+        system = GetComponent<ParticleSystem>();
+    }
+    void Start()
+    {
+        characterHealth.HealthChanged += playHitEffect;
+    }
+
+    private void playHitEffect()
+    {
+        //When players health changes play damage partice effect
+        if (system != null)
```

Comment:
If you require component, you might skip check for null.

Reply:
Ok will make this change.